### PR TITLE
Updated readme with useful info

### PR DESCRIPTION
**Pull request by mkolesnik**
Updated README with some useful information and code examples

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,63 @@
...
+* Cleaning up the cloud after Submariner has been uninstalled.
+
+These capabilities aim to be idempotent, so in case of failure or other necessity they are safe to re-run.
+
+The API defines a `Reporter` type which has the capability to report on the latest operation performed in the cloud.
+
+### Prepare a cloud for Submariner
+
+Preparing a cloud for Submariner takes in number of gateways, internal ports for intra-cluster communications (Submariner components)
+and public ports for inter-cluster communications (Submariner to Submariner).
```
Comment (tpantelis) [inline README.md]:
```suggestion
The `PrepareForSubmarinerInput` function takes the number of gateways, the internal ports used for intra-cluster communication between Submariner components,
and the public ports used for inter-cluster communication between Submariner gateways.
```

File: README.md [hunk 2]
```diff
@@ -1 +1,63 @@
...
+		},
+		Gateways: gateways,
+	}
+	err := cloud.PrepareForSubmariner(input, reporter)
+
+```
+
+### Clean up a cloud after Submariner has been uninstalled
+
+Cleanup doesn't take input as it finds all preparation work done by the library and reverses it.
```
Comment (tpantelis) [inline README.md]:
```suggestion
The `CleanupAfterSubmariner` function reverses all the preparation work previously done by the library.
```

File: README.md [hunk 3]
```diff
@@ -1 +1,63 @@
...
+
+```go
+	err := cloud.CleanupAfterSubmariner(repoter)
```
Comment (tpantelis) [inline README.md]:
```suggestion
	err := cloud.CleanupAfterSubmariner(reporter)
```

File: README.md [hunk 4]
```diff
@@ -1 +1,63 @@
...
+```
+
+## AWS
+
+AWS is supported by the library. In order to prepare an AWS instance, it needs to have OpenShift pre-installed and running.
```
Comment (tpantelis) [inline README.md]:
```suggestion

## Supported Cloud Providers

### AWS

In order to prepare an AWS instance, it needs to have OpenShift pre-installed and running.

```

**Review by tpantelis — CHANGES_REQUESTED** [review]

**Review by tpantelis — APPROVED** [review]

**Comment by nyechiel** [comment]
@mkolesnik 

README.md:23:141 MD013/line-length Line length [Expected: 140; Actual: 157]
README.md:56 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]

**Review by nyechiel — APPROVED** [review]
LGTM, but there are some Markdown failures

**Pull request merged**